src/core/quantum_gate.jl

```diff
@@ -666,6 +746,26 @@ get_operator(gate::Pi8) = pi_8(gate.type)
 
 get_inverse(gate::Pi8) = pi_8_dagger(gate.target[1],gate.type)
 
+"""
+    pi_8_diag(target)
+
+Return a π/8 `Gate` (also known as a ``T`` `Gate`), which applies the [`pi_8_diag()`](@ref) `DiagonalOperator` to the `target` qubit.
+"""
+pi_8_diag(target::Integer) = Pi8_Diag(["T"], "t", target)
+
+struct Pi8_Diag <: AbstractGate
+    display_symbol::Vector{String}
```

Comment:
Display symbol should not be part of the gate. It should be part of what is displaying the gate.